### PR TITLE
[FTRMT032022] Oscar Gonzalez

### PR DESCRIPTION
**Pull request by oscargonzalezdev**
-[X] solve lab-react-ironcontacts

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -4,12 +4,22 @@ import { useState } from 'react';
...
+  // console.log(fiveContacts);
   const [contacts, setContacts] = useState(fiveContacts);
+  const newContactsArr = contactsArr.slice(5);
+  // console.log(newContactsArr);
+  const randomIndex = Math.floor(Math.random() * newContactsArr.length);
+  const newContact = newContactsArr[randomIndex];
 
+  const randomContact = () => {
+    setContacts([...fiveContacts, newContact])
+  }
```
Comment (TA-Remote) [inline src/App.js:17]: To make code more readable, it is always good practice to separate code in clear block/functions, so here I would suggest having lines 9,11,12 inside randomContact function

File: src/App.js [hunk 2]
```diff
@@ -10,15 +10,23 @@ function App() {
...
+  const displayedContacts = [...contacts, newContact]
 
   const randomContact = () => {
-    setContacts([...fiveContacts, newContact])
+   return setContacts(displayedContacts)
+  }
+  const sortByName = () => {
+    const sortedContacts = [...contacts].sort((a, b) => { return (a > b ? 1 : (a === b ? 0 : -1)) })
+    setContacts(sortedContacts);
   }
```
Comment (TA-Remote) [inline src/App.js:21]: Alternative solution:

```
  const sortedContacts = () => {
    setContacts(contacts => [...contacts].sort((a,b) => a.name.localeCompare(b.name)))
  }
```

File: src/App.js [hunk 3]
```diff
@@ -16,16 +16,24 @@ function App() {
...
   const sortByName = () => {
-    const sortedContacts = [...contacts].sort((a, b) => { return (a > b ? 1 : (a === b ? 0 : -1)) })
-    setContacts(sortedContacts);
+    const sortedByName = [...contacts.sort((a, b) => { return (a.name > b.name ? 1 : (a.name === b.name ? 0 : -1)) })]
+    return setContacts(sortedByName);
   }
+  const sortByPopularity = () => {
+    const sortedByPopularity = [...contacts.sort((a, b) => { return (a.popularity > b.popularity ? -1 : (a.popularity === b.popularity ? 1 : 0)) })]
+    return setContacts(sortedByPopularity);
+  }
```
Comment (TA-Remote) [inline src/App.js:25]: You should sort copied array, so have square brackets only around `...contacts`, and not the whole expression

Alternative solution:

```
  const sortedByPopularity = () => {
    setContacts(contacts => [...contacts].sort((a,b) => b.popularity - a.popularity))
  }
```

**Review by TA-Remote — COMMENTED** [review]
Great work on the lab Oscar, keep it up!!

File: src/App.js (hunk 1 above)
Comment (oscargonzalezdev) [reply src/App.js:17]: Okay. I see

File: src/App.js (hunk 3 above)
Comment (oscargonzalezdev) [reply src/App.js:25]: it makes sense :)

**Pull request closed without merging**